libs/indicom.c

```diff
@@ -315,16 +317,39 @@ void IDLog(const char *fmt, ...)
     va_end(ap);
 }
 
+double time_ns()
+{
+struct timespec ts;
+#ifdef __MACH__ // OS X does not have clock_gettime, use clock_get_time
+clock_serv_t cclock;
+mach_timespec_t mts;
+host_get_clock_service(mach_host_self(), CALENDAR_CLOCK, &cclock);
+clock_get_time(cclock, &mts);
+mach_port_deallocate(mach_task_self(), cclock);
+ts.tv_sec = mts.tv_sec;
+ts.tv_nsec = mts.tv_nsec;
+#else
+timespec_get(&ts, TIME_UTC);
+#endif
+return (double)ts.tv_sec+(double)(ts.tv_nsec%1000000000)/1000000000.0;
+}
+
 /* return current system time in message format */
 const char *timestamp()
 {
     static char ts[32];
+    char iso8601[32];
     struct tm *tp;
-    time_t t;
+    struct timespec tm;
+
+    timespec_get(&tm, TIME_UTC);
+    time_t t = (time_t)tm.tv_sec;
+    long n    = tm.tv_nsec % 1000000000;
```

Comment:
the regular timestamp function should not be changed as it would break the logs.

Reply:
this is done